processor/src/main/kotlin/com/syouth/kmapper/processor/base/Utils.kt

```diff
@@ -41,7 +41,10 @@ internal fun KSFunctionDeclaration.getMappingInformation(): MapperInformation {
     val from = parameters[0].type.resolve()
     val to = returnType!!.resolve()
     if (from.declaration !is KSClassDeclaration || to.declaration !is KSClassDeclaration) throw IllegalStateException("Mapping should be from class to class.")
-    if (!from.isDataClass() || !to.isDataClass()) throw IllegalStateException("Mapping is only supported for data classes")
+    if (!(from.isDataClass() && to.isDataClass()) &&
```

Comment:
This should work for all 4 cases (Data<>Data, Data<>Class, Class<>Data, Class<>Class)

Reply:
It does now with my latest commit.
Tested this over at https://github.com/cyc1ingsir/kmapper-examples/blob/482c456b645f66525e2645026d4bb79eeee70f9f/src/main/kotlin/com/syouth/kmapper/testload/Main.kt#L57-L60